bouncer/tests/request_swap_deposit_address_with_affiliates.ts

```diff
@@ -0,0 +1,258 @@
+    'bcrt1qzhzr5p67ukjyzfmxsh53a8rv3p06nqq0m3md3q', // P2WPKH
+    'bcrt1qqazrtdsvdnl8pv6ypz4jv9h7ud0fs5u4tullapvan2amku7eyujsum5wt8', // P2WSH
+    'bcrt1plrgeugjy6vsehsythzxy4jg5ga2zdvuf4zwjchynd78ldklacs4q52p6g5', // Taproot
+  ],
+  Ethereum: ['0xa56A6be23b6Cf39D9448FF6e897C29c41c8fbDFF'],
+  Arbitrum: ['0xa56A6be23b6Cf39D9448FF6e897C29c41c8fbDFF'],
+  Polkadot: ['1yMmfLti1k3huRQM2c47WugwonQMqTvQ2GUFxnU7Pcs7xPo'],
+  Solana: ['3yKDHJgzS2GbZB9qruoadRYtq8597HZifnRju7fHpdRC'],
+} as const;
+
+const entries = Object.entries as <T>(o: T) => [keyof T, T[keyof T]][];
+
+const baseCases = entries(addresses).flatMap(([destChain, addrs]) =>
+  addrs.map(
+    (destAddress) =>
+      ({
+        srcAsset: 'FLIP',
+        srcChain: 'Ethereum',
+        destChain,
+        destAsset: chainConstants[destChain].assets[0],
+        destAddress,
+      }) as NewSwapRequest,
+  ),
+);
+
+const refundCases = entries(addresses).flatMap(([srcChain, addrs]) =>
+  addrs.map(
+    (refundAddress) =>
+      ({
+        destAsset: 'FLIP',
+        destChain: 'Ethereum',
+        destAddress: addresses.Ethereum[0],
+        srcChain,
+        srcAsset: chainConstants[srcChain].assets[0],
+        fillOrKillParams: {
+          refundAddress,
+          minPriceX128: '1',
+          retryDurationBlocks: 100,
+        },
+      }) as NewSwapRequest,
+  ),
+);
+
+const dcaCase: NewSwapRequest = {
+  ...refundCases[0],
+  dcaParams: {
+    numberOfChunks: 7200,
+    chunkIntervalBlocks: 2,
+  },
+};
+
+const withCommission: NewSwapRequest = {
+  ...baseCases[0],
+  commissionBps: 100,
+};
+
+const withAffiliates: NewSwapRequest = {
+  ...baseCases[0],
+  affiliates: [{ account: account2.address, commissionBps: 100 }],
+};
+
+const ccmCase: NewSwapRequest = {
+  ...baseCases.find((c) => c.destChain === 'Arbitrum')!,
+  ccmParams: {
+    message: '0xcafebabe',
+    gasBudget: '1000000',
+    cfParameters: '0xdeadbeef',
+  },
+};
+
+const main = async () => {
+  await using api = await getChainflipApi();
+  let nonce = (await api.rpc.system.accountNextIndex(account.address)).toJSON() as number;
+
+  const allCases = [...baseCases, ...refundCases, dcaCase, withCommission, withAffiliates, ccmCase];
+  const results = await Promise.allSettled(
+    allCases.map((params) => requestSwapDepositAddress(api, params, () => nonce++)),
+  );
+
+  // optimism 🚀
+  let success = true;
+
+  results.forEach((result, i) => {
+    if (result.status === 'fulfilled') {
+      console.log('✅', `request swap deposit address with affiliates ${i} passed`);
+    } else {
+      // realism 😔
+      success = false;
+      console.error(
+        '❌',
+        `request swap deposit address with affiliates ${i} failed`,
+        (result.reason as Error).message,
+        allCases[i],
+      );
+    }
+  });
+
+  if (!success) {
+    console.error('Some tests failed');
+    process.exit(1);
+  }
+};
+
+export const depositChannelCreation = new ExecutableTest('Deposit-Channel-Creation', main, 360);
```

Comment:
Would be great if you can add `depositChannelCreation` to the `commands/run_test.rs` file so it can be ran individually for debugging.

Reply:
done, thanks for the review!